App crashes with `TransactionTooLargeException` when deleting deck with many subdecks

### Checked for duplicates?

- [x] This issue is not a duplicate

### Does it also happen in the desktop version?

- [x] This bug does not occur in the latest version of Anki Desktop

### What are the steps to reproduce this bug?

When attempting to delete a deck that contains many subdecks/children, the app crashes with a `TransactionTooLargeException`. The crash occurs because `disableDeckAndChildrenShortcuts()` tries to disable all deck shortcuts in a single Binder IPC transaction, which exceeds Android's 1MB limit.

## Steps to reproduce:
   1. Create or import a deck with many subdecks (hundreds or thousands)
   2. Long-press on the parent deck to open context menu
   3. Select "Delete deck"
   4. App crashes immediately


### Expected behaviour

The deck should be deleted successfully without crashing.

## Actual behavior:
   App crashes with java.lang.RuntimeException: android.os.TransactionTooLargeException: data parcel size 3555756 bytes



### Debug info

```text
Logcat:

     java.lang.RuntimeException: android.os.TransactionTooLargeException: data parcel size 3555756 bytes
         at android.content.pm.ShortcutManager.disableShortcuts(ShortcutManager.java:388)
         at androidx.core.content.pm.ShortcutManagerCompat.disableShortcuts(ShortcutManagerCompat.java:608)
         at com.ichi2.anki.DeckPicker.disableDeckAndChildrenShortcuts(DeckPicker.kt:2255)
         at com.ichi2.anki.DeckPicker.handleContextMenuSelection(DeckPicker.kt:888)
     Caused by: android.os.TransactionTooLargeException: data parcel size 3555756 bytes
         at android.os.BinderProxy.transactNative(Native Method)
```

### (Optional) Anything else you want to share?

_No response_

### Research

- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

--- a/AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt
@@ -168,6 +168,7 @@ import com.ichi2.anki.ui.animations.fadeIn
 import com.ichi2.anki.ui.animations.fadeOut
 import com.ichi2.anki.ui.windows.permissions.PermissionsActivity
 import com.ichi2.anki.utils.Destination
+import com.ichi2.anki.utils.ShortcutUtils
 import com.ichi2.anki.utils.ext.dismissAllDialogFragments
 import com.ichi2.anki.utils.ext.getSizeOfBitmapFromCollection
 import com.ichi2.anki.utils.ext.setFragmentResultListener
@@ -2249,10 +2250,9 @@ open class DeckPicker :
     /** Disables the shortcut of the deck and the children belonging to it.*/
     @NeedsTest("ensure collapsed decks are also deleted")
     private fun disableDeckAndChildrenShortcuts(did: DeckId) {
-        // Get the DeckId and all child DeckIds
         val deckTreeDids = dueTree?.find(did)?.map { it.did.toString() } ?: listOf()
         val errorMessage: CharSequence = getString(R.string.deck_shortcut_doesnt_exist)
-        ShortcutManagerCompat.disableShortcuts(this, deckTreeDids, errorMessage)
+        ShortcutUtils.disableShortcuts(this, deckTreeDids, errorMessage)
     }
 
     fun renameDeckDialog(did: DeckId) {

--- a/AnkiDroid/src/main/java/com/ichi2/anki/utils/ShortcutUtils.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/utils/ShortcutUtils.kt
@@ -1,0 +1,56 @@
+/*
+ *  Copyright (c) 2025 Snowiee <xenonnn4w@gmail.com>
+ *
+ *  This program is free software; you can redistribute it and/or modify it under
+ *  the terms of the GNU General Public License as published by the Free Software
+ *  Foundation; either version 3 of the License, or (at your option) any later
+ *  version.
+ *
+ *  This program is distributed in the hope that it will be useful, but WITHOUT ANY
+ *  WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+ *  PARTICULAR PURPOSE. See the GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along with
+ *  this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.ichi2.anki.utils
+
+import android.content.Context
+import androidx.core.content.pm.ShortcutManagerCompat
+import com.ichi2.anki.runCatchingWithReport
+import timber.log.Timber
+
+/**
+ * Wrapper around [ShortcutManagerCompat] to handle platform-specific issues.
+ */
+object ShortcutUtils {
+    /**
+     * Batch size for shortcut operations to avoid [android.os.TransactionTooLargeException].
+     * Android's Binder IPC has a 1MB transaction limit.
+     */
+    private const val BATCH_SIZE = 100
+
+    /**
+     * Disables shortcuts by their IDs, handling large lists by batching to avoid
+     * [android.os.TransactionTooLargeException].
+     *
+     * @param context The context to use
+     * @param shortcutIds List of shortcut IDs to disable
+     * @param disabledMessage Message shown when user tries to use a disabled shortcut
+     */
+    fun disableShortcuts(
+        context: Context,
+        shortcutIds: List<String>,
+        disabledMessage: CharSequence,
+    ) {
+        if (shortcutIds.isEmpty()) return
+
+        shortcutIds.chunked(BATCH_SIZE).forEach { batch ->
+            runCatchingWithReport(
+                "Failed to disable shortcuts batch",
+            ) { ShortcutManagerCompat.disableShortcuts(context, batch, disabledMessage) }
+                .onFailure { e -> Timber.w(e, "Failed to disable shortcuts batch") }
+        }
+    }
+}

